drink/src/lib.rs

```diff
@@ -44,13 +45,19 @@ pub type EventRecordOf<Runtime> = EventRecord<
     <Runtime as frame_system::Config>::Hash,
 >;
 
+type ContractBalanceOf<R> =
+    <<R as pallet_contracts::Config>::Currency as Inspect<AccountIdFor<R>>>::Balance;
+
 /// Copied from pallet-contracts.
-pub type ContractInstantiateResultFor<Runtime> =
-    ContractInstantiateResult<AccountIdFor<Runtime>, BalanceOf<Runtime>, EventRecordOf<Runtime>>;
+pub type ContractInstantiateResultFor<Runtime> = ContractInstantiateResult<
+    AccountIdFor<Runtime>,
+    ContractBalanceOf<Runtime>,
+    EventRecordOf<Runtime>,
+>;
 
 /// Copied from pallet-contracts.
 pub type ContractExecResultFor<Runtime> =
-    ContractExecResult<BalanceOf<Runtime>, EventRecordOf<Runtime>>;
+    ContractExecResult<ContractBalanceOf<Runtime>, EventRecordOf<Runtime>>;
```

Comment:
What if a contract makes a native token transfer which type is `BalanceOf` (not `ContractBalanceOf`)?

Reply:
should be the same, will get rid of the type dup